picoharpApp/src/picopeaks.c

```diff
@@ -318,8 +317,8 @@ static void accum_buffer(
 void pico_process_fast(struct pico_data *self)
 {
     self->max_bin = compute_max_bin(self->countsbuffer);
-    self->total_count_fast = compute_total_count(self->countsbuffer);
-    self->turns_fast = 1e-3 * self->current_time * self->turns_per_sec;
+    self->total_count_fast = compute_total_count(self->countsbuffer);       // total_count_fast does not exist in struct pico_data
+    self->turns_fast = 1e-3 * self->current_time * self->turns_per_sec;     // turns_fast does not exist in struct pico_data
```

Comment:
What on earth are you saying here?  If the named fields don't exist ... then this code won't compile.  I assume it does (haven't got that far!)

Reply:
I was wondering, because the struct pico_data doesn't have this members.
https://github.com/DiamondLightSource/picoharp/blob/master/picoharpApp/src/picopeaks.h#L20

Reply:
Ah, there is some unpleasant trickery in the declaration of `struct pico_data`: the macro `DECLARE_RANGE` actually adds arrays `total_count_fast` together with `total_count_5` ... `..._60` to `_all`.  Nasty stuff, pretty sure I didn't write it!